tests/test_discretisation/test_arbitrary_discretiser.py

```diff
@@ -34,3 +36,35 @@ def test_arbitrary_discretiser():
     )
     X = transformer.fit_transform(data)
     pd.testing.assert_frame_equal(X, data_t1)
+
+
+def test_error_if_input_df_contains_na_in_transform(df_vartypes, df_na):
```

Comment:
I would rename this test to
`test_error_when_nan_introduced_during_transform()`

Because like with the encoders, the transformer should raise an error 1) if the data to transform has NA or 2) when NA are introduced by the transformation itself (the latter is the one you are working on)